### PR TITLE
UIIN-1718 create Jest/RTL test for withLocation.js

### PR DESCRIPTION
**Pull request by cchitneedi**
Refs:
Ticket: [UIIN-1718]
URL: https://issues.folio.org/browse/UIIN-1718

### CODE REVIEW

**Review by ssandupatla — APPROVED** [review]

File: src/withLocation.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    search: '?filters=test1&query=test2&sort=test3&qindex=test',
+    pathname: '/testPathName',
+  },
+  history: {
+    push: jest.fn()
+  },
+  updateLocation: mockUpdateLocation,
+  goTo: mockGoto,
+  getParams: mockGetParams,
+  getSearchParams: mockGetSearchParams
```
Comment (zburke) [inline src/withLocation.test.js:21]: These mocks are only used here so there's no reason to define them separately/globally. Just assign them the way you do `defaultProps.history.push`.

File: src/withLocation.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    };
+    const Location = withLocation(WrappedComponent, { ...defaultProps });
+    render(<MemoryRouter><Location /></MemoryRouter>);
+  });
+  afterEach(() => {
+    cleanup();
+  });
+  it('updateLocation should be called', async () => {
+    const button = screen.getByTestId('updateLocation');
+    expect(fireEvent.click(button)).toBe(true);
```
Comment (zburke) [inline src/withLocation.test.js:49]: As noted in the [documentation for fireEvent](https://testing-library.com/docs/dom-testing-library/api-events), "you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro)." Please do.

**Review by zburke — CHANGES_REQUESTED** [review]
Please use the functions from `user-event` in place of `fireEvent()` unless you have a specific reason not to.

File: src/withLocation.test.js (hunk 1 above)
Comment (cchitneedi) [reply src/withLocation.test.js]: changes are done as you suggested.

**Review by KetineniM — APPROVED** [review]

**Review by VSnehalatha — APPROVED** [review]

**Review by ssandupatla — APPROVED** [review]

**Review by mkuklis — APPROVED** [review]

**Pull request merged**